pkg/query-service/app/server.go

```diff
@@ -305,6 +308,65 @@ func loggingMiddleware(next http.Handler) http.Handler {
 	})
 }
 
+func LogCommentEnricher(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		referrer := r.Header.Get("Referer")
+
+		var path, dashboardID, alertID, page, client, viewName, tab string
+
+		if referrer != "" {
+			referrerURL, _ := url.Parse(referrer)
+			client = "browser"
+			path = referrerURL.Path
+
+			if strings.Contains(path, "/dashboard") {
+				// Split the path into segments
+				pathSegments := strings.Split(referrerURL.Path, "/")
+				// The dashboard ID should be the segment after "/dashboard/"
+				// Loop through pathSegments to find "dashboard" and then take the next segment as the ID
+				for i, segment := range pathSegments {
+					if segment == "dashboard" && i < len(pathSegments)-1 {
+						// Return the next segment, which should be the dashboard ID
+						dashboardID = pathSegments[i+1]
+					}
+				}
+				page = "dashboards"
+			} else if strings.Contains(path, "/alerts") {
+				urlParams := referrerURL.Query()
+				alertID = urlParams.Get("ruleId")
+				page = "alerts"
+			} else if strings.Contains(path, "logs") && strings.Contains(path, "explorer") {
+				page = "logs-explorer"
+				viewName = referrerURL.Query().Get("viewName")
+			} else if strings.Contains(path, "/trace") || strings.Contains(path, "traces-explorer") {
+				page = "traces-explorer"
+				viewName = referrerURL.Query().Get("viewName")
+			} else if strings.Contains(path, "/services") {
+				page = "services"
+				tab = referrerURL.Query().Get("tab")
+				if tab == "" {
+					tab = "OVER_METRICS"
+				}
+			}
+		} else {
+			client = "api"
```

Comment:
`api` client calls would not have any log information?

Reply:
It's just a query range API call from user code so no other info can be added.